http/check.go

```diff
@@ -3,16 +3,53 @@ package http
 import (
 	"crypto/tls"
 	"fmt"
-	"github.com/flanksource/canary-checker/pkg"
 	"io/ioutil"
 	"log"
 	"net"
 	"net/http"
 	"net/url"
 	"strconv"
 	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/flanksource/canary-checker/pkg"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	opsProcessed = prometheus.NewCounter(prometheus.CounterOpts{
```

Comment:
Can we decouple the check from prometheus via the api.Metric ? 

Ops can be inferred, and if we add Duration to  https://github.com/flanksource/canary-checker/blob/master/pkg/api.go#L28 then it leaves just status_code and dnsFailed as custom metrics